### DISCUSSION TITLE
KeyboardInterrupt while using --reload on Windows

### DISCUSSION THREAD

**Discussion by PratikTayshete** [fastapi/fastapi#11617] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I searched the FastAPI documentation, with the integrated search.
- [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI


app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Root"}
```

### Description

Running the FastAPI application using uvicorn with the `--reload` argument. Anytime I make any changes within the script and save the file it throws a `KeyboardInterrupt` while restarting the application.

The command that I use is as follows:
`uvicorn app:app --reload`

```
KeyboardInterrupt
INFO:     Started server process [11708]
INFO:     Waiting for application startup.
INFO:     Application startup complete.

```

### Operating System

Windows

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

 3.9.19

### Additional Context

I am using anaconda to create the python environment. The version is `conda 4.9.2`

**Comment by YuriiMotov** [comment, marked as answer]
Hi!
There is a known issue in Uvicorn v0.29 (https://github.com/encode/uvicorn/issues/2294).
As a workaround you can downgrade Uvicorn to 0.28.1

**Reply by PratikTayshete** [reply]
Thank you @YuriiMotov I downgraded uvicorn to 0.28.1 in Windows and it is not showing the KeyboardInterrupt issue now.

**Reply by evnchn** [reply]
Note: Uvicorn v0.30.0 has addressed this issue. 

https://github.com/encode/uvicorn/releases/tag/0.30.0

> Suppress side effects of signal propagation (https://github.com/encode/uvicorn/pull/2317)
